crates/chain/src/local_chain.rs

```diff
@@ -2,15 +2,166 @@
 
 use core::convert::Infallible;
 
-use alloc::collections::BTreeMap;
+use crate::collections::BTreeMap;
+use crate::{BlockId, ChainOracle};
+use alloc::sync::Arc;
 use bitcoin::BlockHash;
 
-use crate::{BlockId, ChainOracle};
+/// A structure that represents changes to [`LocalChain`].
+pub type ChangeSet = BTreeMap<u32, Option<BlockHash>>;
```

Comment:
Am I right in saying that if the changeset contains, for example, `(32, None)`, this would mean "the block with height 32 doesn't exist anymore"?

Reply:
Correct!

Reply:
Yeah we probably need to include this in the docs.

Reply:
No need, I'll do it here. Probably need to rebase anyway for the other suggested changes.